packages/multi-select-combo-box/src/vaadin-multi-select-combo-box-internal.js

```diff
@@ -246,6 +269,54 @@ class MultiSelectComboBoxInternal extends ComboBoxDataProviderMixin(ComboBoxMixi
       );
     }
   }
+
+  /**
+   * Override method inherited from the combo-box
+   * to render only selected items when read-only,
+   * even if a different set of items is provided.
+   *
+   * @protected
+   * @override
+   */
+  _setOverlayItems(items) {
+    const effectiveItems = this.readonly ? this.selectedItems : items;
+    super._setOverlayItems(effectiveItems);
+  }
+
+  /**
+   * Override method inherited from the combo-box
+   * to not request data provider when read-only.
+   *
+   * @param {number}
+   * @return {boolean}
+   * @protected
+   * @override
+   */
+  _shouldLoadPage(page) {
+    if (this.readonly) {
+      return false;
+    }
+
+    return super._shouldLoadPage(page);
+  }
+
+  /** @private */
+  _readonlyChanged(readonly, oldReadonly) {
+    if (readonly) {
+      this.__savedItems = this._getOverlayItems();
+      this._setOverlayItems(this.selectedItems);
+    } else if (oldReadonly) {
+      this._setOverlayItems(this.__savedItems);
+      this.__savedItems = null;
+    }
+  }
+
+  /** @private */
+  _readonlyItemsChanged(readonly, selectedItems) {
+    if (readonly && selectedItems) {
+      this._setOverlayItems(selectedItems);
+    }
```

Comment:
This was previously covered by the host `_selectedItemsChanged` observer, but the test was missing.